src/object_.d

```diff
@@ -169,8 +169,11 @@ bool opEquals(Object lhs, Object rhs)
     if (lhs is null || rhs is null) return false;
 
     // If same exact type => one call to method opEquals
-    if (typeid(lhs) is typeid(rhs) || typeid(lhs).opEquals(typeid(rhs)))
+    if (!__ctfe && // CTFE doesn't like comparing typeid
```

Comment:
Originally, there was only one identity comparison `typeid(lhs) is typeid(rhs)`. As far as I know, `typeid(lhs).opEquals(typeid(rhs))` was added for the DLL/shared library support. So I think following code would be better:

```d
    // If same exact type => one call to method opEquals
    if (typeid(lhs) is typeid(rhs) ||
        !__ctfe && typeid(lhs).opEquals(typeid(rhs)))
    {
        return lhs.opEquals(rhs);
    }
```

During CTFE interpretation. all typeid(T) objects are unique. So the code won't create difference between compile-time and runtime.